### ISSUE TITLE
How to investigate slow saves for complex entries ?

### ISSUE THREAD

**Issue by loic-marin** [spicywebau/craft-neo#733]
### What question would you like to ask?

Hi everyone, I have a question about slow saves on our entries.

We're building a new project and we needed to generate a behavior close to a page builder.
In order to do that we created a dozen matrix blocks that we inserted in a neo field.
The behavior itself works fine but as the number of blocks on the page grow, the save time also grow accordingly.
For a somewhat normal page, we end up with a loading time of 10 seconds and more to save the entry.
If I create a bigger page, I end up with 20-30 seconds.
I checked a lot of discussion here :
https://github.com/craftcms/cms/issues/9802
https://craftcms.stackexchange.com/questions/39616/debugging-slow-loading-entries-with-matrix-field-in-control-panel
https://github.com/spicywebau/craft-neo/issues/496

From what I understand, the performance issues have been fixed along the way (we're using Craft 4.3.5 and craft-neo 3.7.0).
I'm not sure how to handle this situation. Has anyone ever run into this kind to behavior ?

Thanks for your help !

**Comment by ttempleton** [comment]
Yeah, the performance issues should be fixed with Craft 4 / Neo 3, by duplicating Matrix/Neo block ownership instead of creating a whole new block when no block content has changed (e.g. when creating/updating drafts, and across revisions). That said, I could imagine some cases where entry saving could still take a little while, but 30 seconds definitely seems excessive.

I have some questions, to try to determine how/where anything could be going wrong, and whether it's Neo-specific. If anyone else reading this is also experiencing performance issues, please feel free to jump in with your own feedback.

- Do you have Craft's `autosaveDrafts` setting disabled?
- Does this occur only when creating a new entry, or on all entry saves?
- Does this also occur with just Matrix blocks, on entries that don't use a Neo field?
- If you use Super Table, does this occur when Matrix blocks have nested Super Table rows (again on entries that don't use a Neo field)?

**Comment by loic-marin** [comment]
Hi @ttempleton,
I'm sorry, I did not have access to a computer for two weeks.
I wanted to make sure to test everything before answering to you : 

- Do you have Craft's autosaveDrafts setting disabled?
I confirm that autosaveDrafts  is disabled

- Does this occur only when creating a new entry, or on all entry saves?
It occurs in both case, new entry and entry saves.

- Does this also occur with just Matrix blocks, on entries that don't use a Neo field?
It does occur with matrix blocks that are using super tables only.

- If you use Super Table, does this occur when Matrix blocks have nested Super Table rows (again on entries that don't use a Neo field)?
We realised that the problem does not occur if we only use matrix block (without Neo and without Supertable).

Also, we realised that this is related to the PHP memory_limit. If we increase this limit, the save is faster but we won't be able to do that on a live server.

Thanks again for your help and sorry for the delay in my response.

**Comment by ttempleton** [comment]
No worries @loic-marin, thanks for that info. With what you've stated regarding the slowdown with Matrix -> Super Table, it does seem like the general issue occurs when there are nested block element fields. I'd recommend [opening a discussion on the Craft repository](https://github.com/craftcms/cms/discussions/new/choose) about this in case they have any insights on what could be improved in Craft itself, but happy to try and help however possible from a Neo-specific perspective.
